src/component/recipeAddForm/RecipeFormModal.tsx

```diff
@@ -39,51 +50,63 @@ const labelText = {
   servings: "Servings",
 };
 
-const HANDLE_INPUT = "handle_input";
-const HANDLE_INGREDIENT_INPUT = "handle_ingredient_input";
+interface RecipeFormModalProps {
+  onClose: () => void;
+}
 
-function RecipeFormModal({ onClose }) {
+function RecipeFormModal({ onClose }: RecipeFormModalProps) {
   const [isLoading, setIsLoading] = useState(false);
   const [errorMessage, setErrorMessage] = useState("");
   const [successMessage, setSuccessMessage] = useState("");
   const [recipeFormState, dispatch] = useReducer(
     recipeFormReducer,
     initialFormValue
   );
-  const { navigate } = useContext(NavigationContext);
-  const { createBookmark } = useContext(BookmarksContext);
-
-  const handleChange = (e) => {
-    dispatch({
-      type: e.target.name ? HANDLE_INPUT : HANDLE_INGREDIENT_INPUT,
-      field: e.target.name || e.target.id,
-      payload: e.target.value,
-    });
+  const { navigate } = useNavigationContext();
+  const { createBookmark } = useBookmarksContext();
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.name) {
+      dispatch({
+        type: RecipeFormActionKind.HANDLE_INPUT,
+        field: e.target.name,
+        payload: e.target.value,
+      });
+    } else {
+      dispatch({
+        type: RecipeFormActionKind.HANDLE_INGREDIENT_INPUT,
+        field: Number(e.target.id),
+        payload: e.target.value,
+      });
+    }
   };
 
   const handleSuccess = () => {
     setIsLoading(false);
     setSuccessMessage("Recipe was successfully uploaded :)");
   };
 
-  const handleError = (err) => {
+  const handleError = (err: Error) => {
     setIsLoading(false);
     setErrorMessage(err.message);
   };
 
-  const handleSubmit = async (e) => {
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
     try {
-      setErrorMessage(null);
+      setErrorMessage("");
       e.preventDefault();
       setIsLoading(true);
-      const recipe = makeRecipe(recipeFormState);
+      const recipe: Omit<RecipeDetailsType, "id"> = {
```

Comment:
Can we create a type definition for `Omit<RecipeDetailsType, "id">`? I think it's getting repeated.